### PR TITLE
Parse user directory on nogui CLI.

### PR DESCRIPTION
**Pull request by vladfi1**
This makes nogui actually usable! Split from #4994.

### CODE REVIEW

File: Source/Core/DolphinWX/MainNoGUI.cpp [hunk 1]
```diff
@@ -380,7 +380,7 @@ static Platform* GetPlatform()
 
 int main(int argc, char* argv[])
 {
-  std::string boot_filename;
+  std::string boot_filename, user_directory;
```
Comment (lioncash) [inline Source/Core/DolphinWX/MainNoGUI.cpp]: This variable should be moved nearest to its use. In this case, above the if statement down below.

Comment (vladfi1) [inline Source/Core/DolphinWX/MainNoGUI.cpp]: Shall I do the same with the `boot_filename`?

Comment (lioncash) [inline Source/Core/DolphinWX/MainNoGUI.cpp]: Sure

Comment (vladfi1) [inline Source/Core/DolphinWX/MainNoGUI.cpp]: done

**Pull request merged**